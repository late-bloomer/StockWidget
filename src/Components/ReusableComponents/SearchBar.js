import React from "react";
import Chip from "@mui/material/Chip";
import Autocomplete from "@mui/material/Autocomplete";
import TextField from "@mui/material/TextField";
import Button from "@mui/material/Button";
import SearchIcon from "@mui/icons-material/Search";
import Paper from "@mui/material/Paper";
import { withStyles } from "@mui/styles";

const styles = (theme) => ({
  searchBar: {
    display: "flex",
    width: "100%",
    flexDirection: "column",
  },
  paperNoOpt: {
    padding: "10px",
  },
  paperNoOptDiv: {
    padding: "10px",
    visibility: "hidden",
  },
});

function SearchBar(props) {
  const {
    classes = {},
    handleOnChangeAutoComplete = () => {},
    handleText = () => {},
    symbolList = [],
    open = false,
    isNoOptions = false,
    onSearchButtonClick = () => {},
  } = props;

  return (
    <div className={classes.searchBar}>
      <Autocomplete
        multiple
        open={open}
        freeSolo
        id="tags-outlined"
        onChange={(event, newValue, reason) => {
          handleOnChangeAutoComplete(newValue, reason);
        }}
        options={symbolList.map((option) => option["1. symbol"].toUpperCase())}
        defaultValue={[]}
        filterSelectedOptions
        clearOnBlur
        noOptionsText={"No Option Available."}
        renderInput={(params) => (
          <TextField
            {...params}
            variant="outlined"
            label="Search Symbol"
            placeholder="Enter Symbol"
            onChange={(event) => {
              handleText(event.target.value);
            }}
          />
        )}
        renderTags={(value, getTagProps) =>
          value.map((option, index) => (
            <Chip variant="filled" label={option} {...getTagProps({ index })} />
          ))
        }
        style={{ width: "100%" }}
      />
      {isNoOptions ? (
        <Paper className={classes.paperNoOpt} elevation={3}>
          No Symbol Available..
        </Paper>
      ) : (
        <div className={classes.paperNoOptDiv}>NA..</div>
      )}
      {/* <Button
        variant="contained"
        onClick={onSearchButtonClick}
        endIcon={<SearchIcon />}
        style={{ marginLeft: "16px" }}
      >
        Search
      </Button> */}
    </div>
  );
}
export default withStyles(styles)(SearchBar);
