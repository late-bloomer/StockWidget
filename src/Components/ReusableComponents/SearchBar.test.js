import React from "react";
import ReactDOM from "react-dom";
import Enzyme, { shallow, render, mount } from "enzyme";
import SearchBar from "./SearchBar";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchBar Component", () => {
  it("should render SearchBar Successfully", () => {
    const wrapper = shallow(<SearchBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
