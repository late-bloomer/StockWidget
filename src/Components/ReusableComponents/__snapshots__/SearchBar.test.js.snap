// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render SearchBar Successfully 1`] = `
<SearchBar
  classes={
    Object {
      "paperNoOpt": "SearchBar-paperNoOpt-2",
      "paperNoOptDiv": "SearchBar-paperNoOptDiv-3",
      "searchBar": "SearchBar-searchBar-1",
    }
  }
/>
`;
